src/dune/exe_rules.ml

```diff
@@ -90,19 +90,33 @@ let executables_rules ~sctx ~dir ~expander ~dir_contents ~scope ~compile_info
      stubs, we link the corresponding object files directly. It would be nice to
      make the code more uniform. *)
   let ext_lib = ctx.lib_config.ext_lib in
+  let link_flags_for_relative_foreign_archives () =
+    Command.Args.S
+      (let libs =
+         Result.ok_exn (Lazy.force (Lib.Compile.requires_link compile_info))
+       in
+       List.map libs ~f:(fun lib ->
+           let archive_paths = Lib_info.foreign_archives (Lib.info lib) in
+           Command.Args.S
```

Comment:
Are we making sure to appropriate dependencies here?

Reply:
Oh, well-spotted! Added dependencies on the archive files in the upcoming commit.